Dockerfile

```diff
@@ -1,151 +1,114 @@
 ARG ARCH=amd64
-ARG NPM_VERSION=6.9.0
-
-# The node version here should match the version of the runtime image which is
-# specified in the base-image subdirectory in the project
-FROM balenalib/raspberry-pi-node:10-run as rpi-node-base
-FROM balenalib/armv7hf-node:10-run as armv7hf-node-base
-FROM balenalib/aarch64-node:10-run as aarch64-node-base
-RUN [ "cross-build-start" ]
-RUN sed -i '/security.debian.org jessie/d' /etc/apt/sources.list
-RUN [ "cross-build-end" ]
-
-FROM balenalib/amd64-node:10-run as amd64-node-base
-RUN echo '#!/bin/sh\nexit 0' > /usr/bin/cross-build-start && chmod +x /usr/bin/cross-build-start \
-	&& echo '#!/bin/sh\nexit 0' > /usr/bin/cross-build-end && chmod +x /usr/bin/cross-build-end
-
-FROM balenalib/i386-node:10-run as i386-node-base
-RUN echo '#!/bin/sh\nexit 0' > /usr/bin/cross-build-start && chmod +x /usr/bin/cross-build-start \
-	&& echo '#!/bin/sh\nexit 0' > /usr/bin/cross-build-end && chmod +x /usr/bin/cross-build-end
-
-FROM balenalib/i386-nlp-node:6-jessie as i386-nlp-node-base
-RUN echo '#!/bin/sh\nexit 0' > /usr/bin/cross-build-start && chmod +x /usr/bin/cross-build-start \
-	&& echo '#!/bin/sh\nexit 0' > /usr/bin/cross-build-end && chmod +x /usr/bin/cross-build-end
-
-# Setup webpack building base images
-# We always do the webpack build on amd64, cause it's way faster
-FROM amd64-node-base as rpi-node-build
-FROM amd64-node-base as amd64-node-build
-FROM amd64-node-base as armv7hf-node-build
-FROM amd64-node-base as aarch64-node-build
-FROM amd64-node-base as i386-node-build
-FROM balenalib/amd64-node:6-build as i386-nlp-node-build
-##############################################################################
-
-FROM $ARCH-node-build as node-build
+ARG NODE_VERSION=10.18.0
 
+FROM balenalib/$ARCH-alpine-supervisor-base:3.11 as BUILD
 
-WORKDIR /usr/src/app
-
-RUN apt-get update \
-	&& apt-get install -y \
-		g++ \
-		git \
-		libsqlite3-dev \
-		make \
-		python \
-		rsync \
-		curl \
-	&& rm -rf /var/lib/apt/lists/
-
-COPY package.json package-lock.json /usr/src/app/
-
-ARG NPM_VERSION
-# We first ensure that every architecture has an npm version
-# which can do an npm ci, then we perform the ci using this
-# temporary version
-RUN curl -LOJ https://www.npmjs.com/install.sh && \
-	# This is required to avoid a bug in uid-number
-	# https://github.com/npm/uid-number/issues/7
-	npm config set unsafe-perm true && \
-	npm_install="#{NPM_VERSION}" npm_config_prefix=/tmp sh ./install.sh && \
-	JOBS=MAX /tmp/bin/npm ci --no-optional --unsafe-perm
-
-COPY webpack.config.js fix-jsonstream.js hardcode-migrations.js tsconfig.json tsconfig.release.json /usr/src/app/
-COPY src /usr/src/app/src
-COPY test /usr/src/app/test
-COPY typings /usr/src/app/typings
-
-RUN npm run test-nolint \
-	&& npm run build
-
-##############################################################################
-
-# Build nodejs dependencies
-FROM $ARCH-node-base as node-deps
+ARG ARCH
+ARG NODE_VERSION
```

Comment:
redefined from above (line 2), intended?

Reply:
Yeah, we need to bring it into the stage, it's why it's redefined again in the latter stage

Reply:
I missed that line, thanks for clarifying